manager/runner_test.go

```diff
@@ -780,6 +780,74 @@ func TestRunner_Start(t *testing.T) {
 		}
 	})
 
+	// Exec would run before template rendering if Wait was defined.
+	t.Run("exec-wait", func(t *testing.T) {
+		t.Parallel()
+
+		testConsul.SetKVString(t, "exec-wait-foo", "foo")
+
+		firstOut, err := ioutil.TempFile("", "foo")
+		if err != nil {
+			t.Fatal(err)
+		}
+		os.Remove(firstOut.Name())       // remove ioutil created file
+		defer os.Remove(firstOut.Name()) // remove template created file
+
+		c := config.DefaultConfig().Merge(&config.Config{
+			Consul: &config.ConsulConfig{
+				Address: config.String(testConsul.HTTPAddr),
+			},
+			Wait: &config.WaitConfig{
+				Min: config.TimeDuration(5 * time.Millisecond),
+				Max: config.TimeDuration(10 * time.Millisecond),
+			},
+			Exec: &config.ExecConfig{
+				// `cat filename` would fail if template hadn't rendered
+				Command: config.String(`cat ` + firstOut.Name()),
```

Comment:
Neat 😄. Took me a second to realize that this is the vital part of the test but totally makes sense. Are there any cross-platform concerns here? Not sure we can really get around it given we're testing `exec`...

Reply:
The `cat` command is POSIX standard which is about as much as I think is possible for cross-platform support.

Reply:
Yeah makes sense.